modules/oe_theme_inpage_navigation/src/Plugin/Condition/InPageNavigationState.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_inpage_navigation\Plugin\Condition;
+
+use Drupal\Core\Condition\ConditionPluginBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\node\NodeInterface;
+use Drupal\oe_theme_inpage_navigation\InPageNavigationHelper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a configurable 'In-page Navigation State' condition.
+ *
+ * This condition checks if the in-page navigation is enabled for the node.
+ *
+ * @Condition(
+ *   id = "oe_theme_inpage_navigation_state",
+ *   label = @Translation("In-page navigation"),
+ *   context_definitions = {
+ *     "node" = @ContextDefinition("entity:node", required = FALSE)
+ *   }
+ * )
+ */
+class InPageNavigationState extends ConditionPluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'inpage_navigation_state' => FALSE,
+    ] + parent::defaultConfiguration();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form['inpage_navigation_state'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Inpage navigation'),
+      '#default_value' => $this->configuration['inpage_navigation_state'],
+      '#options' => [$this->t('Disabled'), $this->t('Enabled')],
+      '#description' => $this->t('Choose with which inpage navigation state this condition should be met.'),
+    ];
+    return parent::buildConfigurationForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    if ($form_state->hasValue('inpage_navigation_state')) {
+      $this->configuration['inpage_navigation_state'] = $form_state->getValue('inpage_navigation_state');
+    }
+    parent::submitConfigurationForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function evaluate() {
+    $node = $this->getContextValue('node');
+
+    if (!$node instanceof NodeInterface) {
+      return $this->isNegated();
+    }
+
+    return InPageNavigationHelper::isInPageNavigationEnabled($node) === $this->configuration['inpage_navigation_state'];
```

Comment:
Here you do === but from what I see, you are storing 1 or 0 for the state configuration. Also, don''t we care about `isNegated()` here?

Reply:
It is handled in this method: \Drupal\Core\Condition\ConditionManager::execute

Reply:
OK, so if `ConditionManager::execute` handles that, why if there is no Node available, we return `isNegated()`? Shouldn't we return FALSE and if it is negated, the condition manager would return true?

Reply:
I expected that if we don't have an available node in context, the conditional plugin will return always FALSE.
If we return just FALSE, evaluate() method return TRUE in this test case: `enabled in-page navigation, negative, no node`.  
I used isNegated() method to inforce `ConditionManager::execute` method always return FALSE.

Reply:
ok